libs/agno/agno/agent/agent.py

```diff
@@ -3175,9 +3146,21 @@ def _make_memories_and_summaries(
 
             # Create session summary
             if self.enable_session_summaries:
+                from copy import deepcopy
+
                 log_debug("Creating session summary.")
+                if self.summary_manager is None:
+                    self.summary_manager = SessionSummarizer(model=deepcopy(self.model))
+                # Set the model on the summary_manager if it is not set
+                elif self.summary_manager.model is None:
+                    self.summary_manager.model = deepcopy(self.model)
                 futures.append(
-                    executor.submit(self.memory.create_session_summary, session_id=session_id, user_id=user_id)  # type: ignore
+                    executor.submit(
+                        self.agent_session.create_session_summary,
```

Comment:
Wouldn't a better pattern be to call the summary manager (or session_summarizer) with the session and get a session_summary out, which you then set on the session? 
This feels too loaded.